tensorboard/plugins/histogram/summary.py

```diff
@@ -124,27 +128,54 @@ def op(name,
 
   Returns:
     A TensorFlow summary op.
+
+  Raises:
+    ValueError: If data is not rank 1 when `edges` is specified.
```

Comment:
This API is really confusing. Reusing the `data` parameter to mean two different things—the actual data, or pre-computed bucket heights—makes it hard to understand what both the API and the implementation are doing.